src/symbolic_memory.ml

```diff
@@ -43,54 +57,62 @@ module M = struct
 
   let blit_string m str ~src ~dst ~len =
     (* Always concrete? *)
-    let src = Int32.to_int @@ concretize_i32 src.e in
-    let dst = Int32.to_int @@ concretize_i32 dst.e in
-    let len = Int32.to_int @@ concretize_i32 len.e in
+    let src = Int32.to_int @@ i32 src in
+    let dst = Int32.to_int @@ i32 dst in
+    let len = Int32.to_int @@ i32 len in
     if
       src < 0 || dst < 0
       || src + len > String.length str
       || dst + len > m.size * page_size
     then Value.Bool.const true
     else begin
       for i = 0 to len - 1 do
-        let byte = Int32.of_int @@ Char.code @@ String.get str (src + i) in
+        let byte = Char.code @@ String.get str (src + i) in
         let dst = Int32.of_int (dst + i) in
-        Hashtbl.replace m.data dst (Value.const_i32 byte)
+        Hashtbl.replace m.data dst (Val (Num (I8 byte)) @: Ty_bitv S8)
       done;
       Value.Bool.const false
     end
 
-  let clone m = { data = Hashtbl.create 0; parent = Some m; size = m.size }
+  let clone m =
+    { data = Hashtbl.create 0
+    ; parent = Some m
+    ; size = m.size
+    ; chunks = Hashtbl.copy m.chunks (* TODO: we can make this lazy as well *)
+    }
 
   let rec load_byte_opt a m =
     match Hashtbl.find_opt m.data a with
     | Some b -> Some b
     | None -> Option.bind m.parent (load_byte_opt a)
```

Comment:
I know this is not code from the current PR, but could you also change it to use `find` and match on `exception Not_found` and `b` directly ?